### DISCUSSION TITLE
Suricata Supression/Documentation with idstools

### DISCUSSION THREAD

**Discussion by Cra5hedC0w** [Security-Onion-Solutions/securityonion#1631] [Unsupported Versions] [answered]
Good Day, 

I am attempting to suppress various SIDs within Suricata. I have reviewed the SO 2.3 docs, and there is no syntax guidance on the "modify" portion of the minion pillars within the SO documentation. What is the syntax for this portion of the pillar? Is there any downside to modifying the suricata rules directly, or is it necessary to go through the idstools pillar/saltstack?

Any guidance/tips are appreciated.

**Comment by TOoSmOotH** [comment, marked as answer]
I think as long as you wrap the line in `'` you should be ok. 
under modified the syntax would be:
```
modify:
  - 're:classtype:trojan-activity "(alert)(.*)" "drop\\2"'
  - 'something else blah'
```

**Reply by Cra5hedC0w** [reply]
Thanks for the assist. Where does "re:classtype:trojan-activity "(alert)(.*)" "drop\\2" come from? Is this something that is in the suricata documentation?  Can you provide some context to this example? My main goal is for something along these lines: 

SID 12345
Exclude Alerts from IP: 10.x.x.x/32 for this SID. 
All other IPs in the subnet, continue to alert on this SID.

**Reply by TOoSmOotH** [reply]
It's from the suricata docs. I would use thresholds for what you are trying to do. https://docs.securityonion.net/en/2.3/managing-alerts.html?highlight=threshold#threshold

**Reply by Cra5hedC0w** [reply]
Excellent. For the pillar file, I don't see one by default. I assume this has to be created from scratch using the template from the docs?

**Reply by TOoSmOotH** [reply]
That is correct. If you drop it in global.sls it will be for all sensors and if you drop it in the minion.sls it will be specific to that sensor.

**Reply by IzacFrank** [reply]
Hello @TOoSmOotH 

Could you please share an example, if possible a screenshot? I have a standalone and I entered like in the pillar example and no success. I tried to disable but still no luck. Now I tried one liner like below. Even disabling them doesn't work. I still get alerts for these host and suricata threshold says "##### The thresholding pillar has not been defined "

This is my pillar file now ~]$ sudo vi /opt/so/saltstack/local/pillar/minions/ids1_standalone.sls

![image](https://user-images.githubusercontent.com/93327361/148207985-003bf07c-7b96-4f5a-90bc-2d3cfe096deb.png)

![image](https://user-images.githubusercontent.com/93327361/148207769-ff844169-08f6-4056-a5c5-2641fa3a0956.png)

Thanks

**Reply by IzacFrank** [reply]
I just like to stop the alerts for specific hosts, maybe suppressing is the wrong approach. Please advise. 
I also tried like below and dash at the beginning of the SIDS.
![image](https://user-images.githubusercontent.com/93327361/148211599-bf5b0945-c979-4dbe-a49e-18490aa2d6fb.png)

Thanks.
Isac

**Reply by dougburks** [reply]
@IzacFrank I've responded to your discussion over at https://github.com/Security-Onion-Solutions/securityonion/discussions/6602. In the future, please avoid replying to older discussions that have already been marked as answered.
